libsql/libsql
# sync cr-sqlite to ext/crr

#147 

copy `cr-sqlite` extension code as-is to `ext/crr`

build & test

https://app.warp.dev/block/FWDFdN9HSg4jaHayk3EwBo
https://app.warp.dev/block/lseqHcw4cpmyqWMBsyexs5

Next steps:
- remove `src/sqlite/shell.c,sqlite3.c,sqlite3.h,sqlite3ext.h` and use the files present in `libsql`
- Integrate w/ autotools
- integrate tests with your CI
- remove reliance on `commit_hook`
- draft new `create crr` syntax
- fuzzing?

Note: the `package.json` and other `js` files are for distribution of this extension on npm.

There are also a number of integration tests in Python and TypeScript (using https://github.com/dubzzz/fast-check) not included here. I'm thinking of the best way to include those.
